src/backend/controllers/users.js

```diff
@@ -0,0 +1,168 @@
+import { generateJWT, hashPassword} from '../utils/authorization.js';
+import model from "../model/users.js";
+
+export async function signup(req, res) {
+
+    if (!req.body.StudentID || !req.body.SchoolEmail || !req.body.Username || !req.body.Fname || !req.body.Lname || !req.body.Password) {
+        return res.status(400).send('Missing value');
+    }
+
+    if (!req.body.SchoolEmail.match(/^[a-zA-Z0-9]+@[a-zA-Z0-9]+\.[A-Za-z.]+$/)) {
```

Comment:
Should this check also check for ntu.edu.tw suffix?

Reply:
OK